chain/store/store.go

```diff
@@ -909,25 +909,30 @@ func (cs *ChainStore) GetRandomness(ctx context.Context, blks []cid.Cid, pers cr
 	defer span.End()
 	span.AddAttributes(trace.Int64Attribute("round", int64(round)))
 
-	//defer func() {
-	//log.Infof("getRand %v %d %d %x -> %x", blks, pers, round, entropy, out)
-	//}()
-	for {
-		nts, err := cs.LoadTipSet(types.NewTipSetKey(blks...))
-		if err != nil {
-			return nil, err
-		}
+	ts, err := cs.LoadTipSet(types.NewTipSetKey(blks...))
+	if err != nil {
+		return nil, err
+	}
 
-		mtb := nts.MinTicketBlock()
+	if round > ts.Height() {
```

Comment:
Yeah, this feels like implementing a bug, unless there is some case when it's meaningful to ask for randomness at height X based on blocks at height Y, with X > Y (I can't think of one).

I'd scrap the second commit, change this PR to merge into `next` (since it'll make us fail to sync painnet), and make sure we include a fix to [this guy](https://github.com/filecoin-project/specs-actors/issues/406) in the reset next week.